### PR TITLE
Some fixes for linear constraints in the history file

### PR DESCRIPTION
**Pull request by nwu63**
## Purpose
This PR fixes an issue with linear constraints being stored in the `conInfo` field of the history file. Since `conNames` uses the information in `conInfo`, the linear constraint name also appears there. Lastly, by default `getValues` returns all values, and will attempt to extract the value of this linear constraint, which does not exist in `funcs` since it's linear.

I updated `getConNames` so that it skips linear constraints. I also updated the part where `conInfo` is written to the history file during optimization, so that linear constraints are no longer being written. Finally, a test is added to check for this.

## Type of change
What types of change is it?
_Select the appropriate type(s) that describe this PR_

- Bugfix (non-breaking change which fixes an issue)

## Testing
I updated tp109 to check that the linear constraint is not stored in either `conInfo` or `conNames`, and that `getValues` no longer throws an error.

## Checklist
_Put an `x` in the boxes that apply._

- [x] I have run `flake8` and `black` to make sure the code adheres to PEP-8 and is consistently formatted
- [x] I have run unit and regression tests which pass locally with my changes
- [x] I have added new tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation

### CODE REVIEW

File: pyoptsparse/pyOpt_history.py [hunk 1]
```diff
@@ -193,10 +193,13 @@ def _processDB(self):
...
+        # metadata
+        self.metadata = self.read("metadata")
+        self.optProb = self.read("optProb")
         # load names
-        self.DVNames = set(self.DVInfo.keys())
-        self.conNames = set(self.conInfo.keys())
-        self.objNames = set(self.objInfo.keys())
+        self.DVNames = set(self.getDVNames())
+        self.conNames = set(self.getConNames())
+        self.objNames = set(self.getObjNames())
```
Comment (marcomangano) [inline pyoptsparse/pyOpt_history.py:203]: Is this also related to py3 update? why were these `.keys()` used when the other functions were available? Anyway, good you updated this

Comment (nwu63) [reply pyoptsparse/pyOpt_history.py:203]: The old implementation used the `.keys()` call, but I realized that I wrote these functions. Previously, these functions did the exact same thing, but I updated `getConNames` to skip linear constraints, and here this should be consistent also, so I switched to using these functions.

**Comment by nwu63** [comment]
> It looks good to go, I am just missing one thing: where are we storing the linear constraints info now? You removed the "extras" from both `conNames` and the history file, not sure about what is left.

The linear constraints are not stored explicitly anywhere, since they are not evaluated in the `obj` or `sens`. However, they are accessible via `optProb.constraints.jac` for linear constraints. Given that they are just mat-vec products, you can just do `A*x` to get the constraint value.

**Comment by marcomangano** [comment]
> The linear constraints are not stored explicitly anywhere, since they are not evaluated in the `obj` or `sens`. However, they are accessible via `optProb.constraints.jac` for linear constraints. Given that they are just mat-vec products, you can just do `A*x` to get the constraint value.

Sounds good, maybe we can add this one liner somewhere in the documentation, if it is not there already and you feel it is worth it.

**Comment by nwu63** [comment]
Agreed, the documentation should explain a bit more on how linear constraints are handled and so forth. I will open an issue for this.
